x-pack/plugins/enterprise_search/public/applications/shared/kibana_breadcrumbs/generate_breadcrumbs.test.ts

```diff
@@ -0,0 +1,151 @@
+    push: jest.fn(),
+  };
+
+  const breadCrumbs = [
+    {
+      text: 'Page 1',
+      path: '/page1',
+    },
+    {
+      text: 'Page 2',
+      path: '/page2',
+    },
+  ];
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const subject = () => appSearchBreadcrumbs(historyMock)(breadCrumbs);
+
+  it('Builds a chain of breadcrumbs with Enterprise Search and App Search at the root', () => {
+    expect(subject()).toEqual([
+      {
+        href: '/',
+        onClick: expect.any(Function),
+        text: 'Enterprise Search',
+      },
+      {
+        href: '/app_search',
+        onClick: expect.any(Function),
+        text: 'App Search',
+      },
+      {
+        href: '/page1',
+        onClick: expect.any(Function),
+        text: 'Page 1',
+      },
+      {
+        href: '/page2',
+        onClick: expect.any(Function),
+        text: 'Page 2',
+      },
+    ]);
+  });
+
+  describe('links', () => {
+    const eventMock = {
+      preventDefault: jest.fn(),
+    };
+
+    it('has a link to Enterprise Search Home page first', () => {
+      subject()[0].onClick(eventMock);
+      expect(historyMock.push).toHaveBeenCalledWith('/');
+    });
+
+    it('has a link to App Search second', () => {
+      subject()[1].onClick(eventMock);
+      expect(historyMock.push).toHaveBeenCalledWith('/app_search');
+    });
+
+    it('has a link to our custom page third', () => {
+      subject()[2].onClick(eventMock);
+      expect(historyMock.push).toHaveBeenCalledWith('/page1');
+    });
+
+    it('has a link to our second custom page last', () => {
+      subject()[3].onClick(eventMock);
+      expect(historyMock.push).toHaveBeenCalledWith('/page2');
+    });
+  });
+});
+
+describe('enterpriseSearchBreadcrumbs', () => {
+  const historyMock = {
+    createHref: jest.fn(),
+    push: jest.fn(),
+  };
+
+  const breadCrumbs = [
+    {
+      text: 'Page 1',
+      path: '/page1',
+    },
+    {
+      text: 'Page 2',
+      path: '/page2',
+    },
+  ];
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const subject = () => enterpriseSearchBreadcrumbs(historyMock)(breadCrumbs);
+
+  it('Builds a chain of breadcrumbs with Enterprise Search at the root', () => {
+    expect(subject()).toEqual([
+      {
+        href: undefined,
+        onClick: expect.any(Function),
+        text: 'Enterprise Search',
+      },
+      {
+        href: undefined,
+        onClick: expect.any(Function),
+        text: 'Page 1',
+      },
+      {
+        href: undefined,
+        onClick: expect.any(Function),
+        text: 'Page 2',
+      },
+    ]);
+  });
+
+  describe('links', () => {
+    const eventMock = {
+      preventDefault: jest.fn(),
+    };
+
+    it('has a link to Enterprise Search Home page first', () => {
+      subject()[0].onClick(eventMock);
+      expect(historyMock.push).toHaveBeenCalledWith('/');
+    });
+
+    it('has a link to our custom page third', () => {
```

Comment:
replace with: `it('has a link to page 1 second')`

Reply:
Updated here: https://github.com/constancecchen/kibana/pull/4/commits/a7b0acd579ac21f46e165ecc49ec1a0643b10d1f